Develappers.BillomatNet.Tests/UnitTests/InvoiceServiceTagsTests.cs

```diff
@@ -2,9 +2,221 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Develappers.BillomatNet.Api.Net;
+using Develappers.BillomatNet.Queries;
+using Develappers.BillomatNet.Tests.UnitTests.Comparer;
+using Develappers.BillomatNet.Types;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+
 namespace Develappers.BillomatNet.Tests.UnitTests
 {
-    public class InvoiceServiceTagsTests : UnitTestBase<ClientService>
+    [SuppressMessage("ReSharper", "StringLiteralTypo")]
+    public class InvoiceServiceTagsTests : UnitTestBase<InvoiceService>
     {
+        [Fact]
+        public async Task GetTagCloud_WithValidData_ShouldReturnCorrectValues()
+        {
+            const string httpResult =
+                "{\"invoice-tags\":{\"invoice-tag\":{\"id\":\"207252\",\"name\":\"Test\",\"count\":\"2\",\"customfield\":\"\"},\"@page\":\"1\",\"@per_page\":\"100\",\"@total\":\"1\"}}";
+            const string expectedUri = "/api/invoice-tags";
+
+            var expectedResult = new List<TagCloudItem>
+            {
+                new TagCloudItem
+                {
+                    Id = 207252,
+                    Name = "Test",
+                    Count = 2
+                }
+            };
+
+            var http = A.Fake<IHttpClient>();
+            A.CallTo(() => http.GetAsync(new Uri(expectedUri, UriKind.Relative), null, A<CancellationToken>.Ignored))
+                .Returns(Task.FromResult(httpResult));
+
+            var sut = GetSystemUnderTest(http);
+            var result = await sut.GetTagcloudAsync();
+
+            A.CallTo(() => http.GetAsync(new Uri(expectedUri, UriKind.Relative), null, A<CancellationToken>.Ignored))
+                .MustHaveHappenedOnceExactly();
+
+            result.TotalItems.Should().Be(1);
+            result.ItemsPerPage.Should().Be(100);
+
+            result.List.Should().Equal(expectedResult, new TagCloudItemEqualityComparer().Equals);
+        }
+
+        [Fact]
+        public async Task GetTagCloudList_WithInvalidCredentials_ShouldThrowNotAuthorizedException()
+        {
+            // arrange
+            var expectedRequestUri = new Uri("/api/invoice-tags", UriKind.Relative);
+            var http = A.Fake<IHttpClient>();
+            A.CallTo(() => http.GetAsync(expectedRequestUri, null, A<CancellationToken>.Ignored))
+                .ThrowsAsync(ExceptionFactory.CreateNotAuthorizedException);
+
+            var sut = GetSystemUnderTest(http);
+
+            //act and assert
+            await Assert.ThrowsAsync<NotAuthorizedException>(() => sut.GetTagcloudAsync());
+            A.CallTo(() => http.GetAsync(expectedRequestUri, null, A<CancellationToken>.Ignored))
+                .MustHaveHappenedOnceExactly();
+        }
+
+
```

Comment:
Develappers.BillomatNet.Tests/UnitTests/InvoiceServiceTagsTests.cs#L73
The code must not contain multiple blank lines in a row. (SA1507)